secure-api-gateway-ob-uk-rs-server/src/main/java/com/forgerock/sapi/gateway/ob/uk/rs/server/api/obie/payment/v4_0_0/vrp/DomesticVrpsApiController.java

```diff
@@ -0,0 +1,288 @@
+import com.forgerock.sapi.gateway.rs.resource.store.repo.entity.payment.FRDomesticVrpPaymentSubmission;
+import com.forgerock.sapi.gateway.rs.resource.store.repo.mongo.payments.DomesticVrpPaymentSubmissionRepository;
+
+import jakarta.servlet.http.HttpServletRequest;
+import lombok.extern.slf4j.Slf4j;
+import uk.org.openbanking.datamodel.v3.vrp.OBDomesticVRPDetailsDataPaymentStatusInnerStatusDetailStatusReason;
+import uk.org.openbanking.datamodel.v3.vrp.OBDomesticVRPResponseDataStatus;
+import uk.org.openbanking.datamodel.v4.common.Meta;
+import uk.org.openbanking.datamodel.v4.common.OBActiveOrHistoricCurrencyAndAmount;
+import uk.org.openbanking.datamodel.v4.common.OBInternalChargeBearerType1Code;
+import uk.org.openbanking.datamodel.v4.vrp.OBDomesticVRPDetails;
+import uk.org.openbanking.datamodel.v4.vrp.OBDomesticVRPDetailsData;
+import uk.org.openbanking.datamodel.v4.vrp.OBDomesticVRPDetailsDataPaymentStatusInner;
+import uk.org.openbanking.datamodel.v4.vrp.OBDomesticVRPDetailsDataPaymentStatusInnerStatusDetail;
+import uk.org.openbanking.datamodel.v4.vrp.OBDomesticVRPRequest;
+import uk.org.openbanking.datamodel.v4.vrp.OBDomesticVRPResponse;
+import uk.org.openbanking.datamodel.v4.vrp.OBDomesticVRPResponseData;
+import uk.org.openbanking.datamodel.v4.vrp.OBDomesticVRPResponseDataChargesInner;
+import uk.org.openbanking.datamodel.v4.vrp.OBInternalPaymentChargeType1Code;
+
+@Controller("DomesticVrpsApiV4.0.0")
+@Slf4j
+public class DomesticVrpsApiController implements DomesticVrpsApi {
+
+    protected static final String DEFAULT_CHARGE_AMOUNT = "1.00";
+    protected static final String DEFAULT_CHARGE_CURRENCY = "GBP";
+
+    private final OBValidationService<OBDomesticVRPRequestValidationContext> paymentRequestValidator;
+    private final DomesticVrpPaymentSubmissionRepository paymentSubmissionRepository;
+    private final PeriodicLimitBreachResponseSimulatorService limitBreachResponseSimulatorService;
+    private final PaymentSubmissionValidator paymentSubmissionValidator;
+    private final RefundAccountService refundAccountService;
+    private final DomesticVRPConsentStoreClient consentStoreClient;
+    private final IdempotentPaymentService<FRDomesticVrpPaymentSubmission, FRDomesticVrpRequest> idempotentPaymentService;
+
+    public DomesticVrpsApiController(
+            DomesticVrpPaymentSubmissionRepository paymentSubmissionRepository,
+            OBValidationService<OBDomesticVRPRequestValidationContext> paymentRequestValidator,
+            @Qualifier(("v4.0.0RestDomesticVRPConsentStoreClient")) DomesticVRPConsentStoreClient consentStoreClient,
+            PeriodicLimitBreachResponseSimulatorService limitBreachResponseSimulatorService,
+            PaymentSubmissionValidator paymentSubmissionValidator,
+            RefundAccountService refundAccountService
+    ) {
+        this.paymentSubmissionRepository = paymentSubmissionRepository;
```

Comment:
null-checks?

Reply:
Added some null messages in case the objects are not ok at this point but in genral this shouldn't happen. As mentioned before, these controllers are copies from the V3.1.10 controllers and we want to keep the same pattern in v4.0